src/Profile/PageProfile.cpp

```diff
@@ -41,7 +41,7 @@ Load(const ProfileMap &map, PageLayout &_pl, const unsigned page)
 {
   char profileKey[32];
   unsigned prefixLen = sprintf(profileKey, "Page%u", page);
-  if (prefixLen <= 0)
+  if (prefixLen == 0)
```

Comment:
While this change is not incorrect, it reduces the robustness of the code (just in case somebody does something stupid and makes the variable signed); it is extremely unlikely that this robustness ever has an effect, but on the other hand, you did not describe any advantage of dropping it, so I must figure that this change is completely pointless.

Reply:
While I think that your argument about robustness is more likely to hide bad code...:

I think the main mistake here is that we check for a bad value and if untrue we execute, 
When we should be checking whether we have a good value and then proceed with execution.

Will make a proposal.

Reply:
Actutally the current state is a bug. 

Sprintf will return a negative value on failure. Which is currently converted to a unsigned value and execution will proceed...

prefixLen should be therefore signed and checked for a positive value.

Reply:
True, the bad code was added by commit 6faa4f7008b
Related question: does anybody know how `sprintf()` can ever fail?